src/scala/org/pantsbuild/zinc/compiler/Main.scala

```diff
@@ -251,6 +331,7 @@ object Main {
       }
     } catch {
       case e: CompileFailed =>
+        dumpDiagnostics(settings, inputs, workingDirectory)
```

Comment:
Why "this thing failed to compile" is communicated through an exception is a mistery to me, but maybe adding a comment saying why it's impossible that we double-dump would be useful. Even better, maybe include the dump in a `finally` block.